Initial Interfaces

mecha opened the pull request.

mecha committed:
Installed dependencies

mecha committed:
Added PHPUnit config for PHPStorm

mecha committed:
Added test for `TransitionEventInterface`

mecha committed:
Added `TransitionEventInterface`

mecha committed:
Changed return type of `getTarget()`

mecha committed:
Removed awareness of current and new states

XedinUnknown commented on src/TransitionEventInterface.php:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @return StateMachineInterface The state machine.
+     */
+    public function getTarget();
```
I'm thinking that maybe leave `getTarget()` as is, and extend [`StateMachineAwareInterface`].

[`StateMachineAwareInterface`]:                         https://github.com/Dhii/machine-state-machine-interface/blob/develop/src/StateMachineAwareInterface.php

mecha committed:
Updated dep `dhii/machine-state-machine-interface`

mecha committed:
`TranstionEventInterface` extends ...

... `StateMachineAwareInterface` rather than overriding the return type for `getTarget()`.
